pages/common/perl.md

```diff
@@ -3,34 +3,18 @@
 > The Perl 5 language interpreter.
 > More information: <https://www.perl.org>.
 
-- Parse and execute a Perl script:
+- Print lines from stdin [m/] matching regex1 and case insensitive [/i] regex2:
```

Comment:
I do not think the modifcation makes the examples easier to comprehend.
I would request, can you try to modify in small steps instead of a mass-rewrite :)

Reply:
Hi, long time Perl developer here :) I'm of course not representative of someone who's never ever used Perl; but I still need to lookup things now and then.

Unfortunately some the previous examples were, IMHO, of very little use ; not even for a someone with little Perl experience as I believe that most users would be looking for how to do "things" and by that I believe that they often look for at least intermediary examples not how to start a command, that they can guess without help.

Apropos help, 'perl --help' answers the 4 first examples and 25 more, I guessed that tldr users know how to use '--help' as they are not complete beginners since they at least know how to install a tldr client. the last example is about getting help, except with '--help' that people already know of.

it's difficult to know who the typical user is; in any case, someone who's going to write some Perl (or other language) would have spend a few minutes in the docs or some sort of online presentation which leaves the people who want to do things without really knowing how, I believe those people are adressed in my changes.

 As for the older examples which did have content, I haven't removed them, just re-worded them for more precision.

I think it's better to read the new version without thinking about the old one, people are afraid of throwing away the baby with the bath water, maybe showing the page to someone who's never seen the older page could give some insight.


Removing the simplistic examples also gives some room for more useful entries.

I spend some time re-writing the page and I did show this to the Perl community before proposing this patch to check I didn't write anything wrong; the only comment I got was about being clearer with the use of the match operator (m).

If you believe that the use of [] to highlight the elements that impact the command make it harder to understand, please tell me what markers should be used.

Reply:
I absolutely concur with the idea of having useful examples.
I myself am not even a beginner with perl. The reason for my comment was, coming from the perspective of would these examples be useful to the overall generic class of people who leverage tldr. I personally  am not qualified to answer this; let's defer to the community, about what their thoughts would be.

In any case, thank you very much for your elaborate reply and contributions :)

Again, welcome!

Reply:
Well I hope you get to learn some Perl, and Bash, those two are fun and incredibly useful.

Since there some room left I will try to find some, I also asked the Perl community on FB but we're all lazy snakes warming in the sun ;)

Perl has, and always had, an excellent documentation and testing, ... and a FAQ, maybe we should pick something from that. https://perldoc.perl.org/perlfaq